build.gradle

```diff
@@ -86,110 +87,90 @@ test {
+          url = 'https://github.com/sonatype-nexus-community/scan-gradle-plugin/issues'
+        }
+        licenses {
+          license {
+            name = 'The Apache Software License, Version 2.0'
+            url = 'http://www.apache.org/licenses/LICENSE-2.0.txt'
+            distribution = 'repo'
           }
-          scm {
-            connection = 'scm:git:git://github.com/sonatype-nexus-community/scan-gradle-plugin.git'
-            developerConnection = 'scm:git:ssh://git@github.com:sonatype-nexus-community/scan-gradle-plugin.git'
-            url = 'https://github.com/sonatype-nexus-community/scan-gradle-plugin'
+        }
+        scm {
+          connection = 'scm:git:git://github.com/sonatype-nexus-community/scan-gradle-plugin.git'
+          developerConnection = 'scm:git:ssh://git@github.com:sonatype-nexus-community/scan-gradle-plugin.git'
+          url = 'https://github.com/sonatype-nexus-community/scan-gradle-plugin'
+        }
+        developers {
+          developer {
+            name = 'Guillermo Varela'
           }
-          developers {
-            developer {
-              name = 'Guillermo Varela'
-            }
-            developer {
-              name = 'Usman Shaikh'
-            }
+          developer {
+            name = 'Usman Shaikh'
           }
         }
-
-        artifactId = project.name
       }
     }
+  }
+  repositories {
+    maven {
+      url = layout.buildDirectory.dir('staging-deploy')
+    }
+  }
+}
 
-    // configure signing if performing release
-    if ('do-release' == System.getenv('CIRCLE_JOB')) {
-      signing {
-        def final encodedKey = System.getenv('SECRING_GPG_ASC_BASE64')
-        def final signingKey = new String(encodedKey.decodeBase64())
-
-        allprojects {
-          println 'setup gpg signing with in-memory key'
-          useInMemoryPgpKeys(
-              signingKey,
-              System.getenv('GPG_PASSPHRASE')
-          )
-        }
+// Plugin publishing configuration
+def pluginUrl = 'https://github.com/sonatype-nexus-community/scan-gradle-plugin'
+pluginBundle {
+  website = pluginUrl
+  vcsUrl = pluginUrl
+  description = pluginDescription
+  tags = ['sonatype', 'scan', 'dependencies', 'ossindex', 'iq server']
+  System.setProperty('gradle.publish.key', System.getenv('GRADLE_PUBLISH_KEY') ?: '')
+  System.setProperty('gradle.publish.secret', System.getenv('GRADLE_PUBLISH_SECRET') ?: '')
+}
 
-        sign publishing.publications.pluginMaven
-        sign publishing.publications.scanPluginMarkerMaven
+jreleaser {
+  signing {
+    active = 'ALWAYS'
+    armored = true
+  }
+  deploy {
+    maven {
+      mavenCentral {
+        sonatype {
+          active = 'ALWAYS'
+          url = 'https://central.sonatype.com/api/v1/publisher'
+          stagingRepository('build/staging-deploy')
+        }
       }
     }
   }
 }
 
+// Configure signing
+signing {
+  useInMemoryPgpKeys(System.getenv('GPG_SIGNING_KEY'), System.getenv('GPG_PASSPHRASE'))
```

Comment:
Will this handle decoding the base64 encoded PGP signing key?

Reply:
I believe the useInMemoryPGPKeys should handle that